pkg/controller/clusterserviceversion/clusterserviceversion_controller.go

```diff
@@ -79,14 +91,35 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 				return !(watchOk && watchLabel == "lite")
 			},
 			DeleteFunc: func(evt event.DeleteEvent) bool {
-				return false
+				_, okAllNamespace := evt.Meta.GetLabels()[allnamespaceTag]
+				watchLabel, watchOk := evt.Meta.GetLabels()[watchTag]
+				_, ignoreOk := evt.Meta.GetAnnotations()[ignoreTag]
+
+				if strings.Contains(evt.Meta.GetName(), utils.CSV_NAME) {
+					return true
+				}
+
+				if ignoreOk {
+					return false
+				}
+
+				if okAllNamespace {
+					return false
+				}
+
+				return !(watchOk && watchLabel == "lite")
 			},
 			CreateFunc: func(evt event.CreateEvent) bool {
 				_, okAllNamespace := evt.Meta.GetLabels()[allnamespaceTag]
+
+				if strings.Contains(evt.Meta.GetName(), utils.CSV_NAME) {
+					return true
+				}
+
 				return !okAllNamespace
 			},
 			GenericFunc: func(evt event.GenericEvent) bool {
-				return false
+				return strings.Contains(evt.Meta.GetName(), utils.CSV_NAME)
```

Comment:
Need to update this to look for our meter definition annotation and queue if it's present.